util/crs-rules-check/rules-check.py

```diff
@@ -711,13 +808,31 @@ def msg(msg):
     else:
         print(msg)
 
+def generate_version_string():
+    """
+    generate version string from git tag
+    program calls "git describe --tags" and converts it to version
+    eg:
+      v4.5.0-6-g872a90ab -> "4.6.0-dev"
+      v4.5.0-0-abcd01234 -> "4.5.0"
+    """
+    result = subprocess.run(["git", "describe", "--tags", "--match", "v*.*.*"], capture_output=True, text=True)
```

Comment:
I prefer `Popen`. But this works, so it's fine.

Reply:
Usually I use `subprocess` - I should keep this as is now.